src/index.ts

```diff
@@ -1,176 +1,102 @@
-        topicMessagesResult = await querySLDTopicMessages(topicId, sequenceNumber);
-        sequenceNumber += topicMessagesResult.length;
-        const sldTopicMessages = topicMessagesResult as SLDTopicMessage [];
-
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        found = sldTopicMessages.filter((message: any) => (message.nameHash.sldHash === nameHash.sldHash.toString('hex')));
-        if (found.length) break;
-      } while (topicMessagesResult.length === API_MAX_LIMIT);
-
-      if (!found.length) throw new Error(`SLD message for:[${nameHash.domain}] not found on topic:[${topicId.toString()}]`);
-      const message = found.reduce(
-        (prev: SLDTopicMessage, curr: SLDTopicMessage) => (prev.nftId > curr.nftId ? prev : curr),
-      );
-      return message;
-    } catch (err) {
-    // eslint-disable-next-line no-console
-      console.log(err);
-      throw new Error('Failed to getMessageByRestQuery');
-    }
-  };
-
-  /**
-   * @description Resolve the owner of a domain by REST queries
-   * @param nameHash: {NameHash} The domain to query
-   * @returns {Promise<AccountId>}
-   */
-  private resolveDomainByRestQuery = async (
-    nameHash: NameHash,
-  ): Promise<AccountId> => {
-    try {
-      const tldTopicMessage = await this.queryTLDTopicMessage(nameHash);
-      const tokenId = TokenId.fromString(tldTopicMessage.tokenId);
-      const topicId = TopicId.fromString(tldTopicMessage.topicId);
 
-      const message = await this.getSLDTopicMessageByHash(nameHash, topicId);
-      const accountId = await queryNFTOwner(message.nftId, tokenId);
+  // Improve method to look for unexpired domains
+  private async getSecondLevelDomain(nameHash: NameHash): Promise<SecondLevelDomain> {
+    const tld: TopLevelDomain = this.getTopLevelDomain(nameHash);
 
-      return accountId;
-    } catch (err) {
-    // eslint-disable-next-line no-console
-      console.log(err);
-      throw new Error('Failed to resolveDomainByRestQuery');
+    for(let offset = 0; ; offset++) {
+      const slds: SecondLevelDomain[] = await this.getSecondLevelDomains(tld.topicId, offset);
+      const sld = slds.find((sld: SecondLevelDomain) => sld.nameHash.sldHash === nameHash.sldHash.toString('hex'));
+      if(sld) return sld;
     }
-  };
 
-  /**
- * @description Resolves a Second Level Domain to the wallet address of the domain's owner
- * @param domain: {string} The domain to query
- * @returns {Promise<AccountId>}
- */
-  resolveSLD = async (
-    domain: string,
-  ): Promise<AccountId> => {
-    try {
-      const nameHash = HashgraphNames.generateNameHash(domain);
-      return await this.resolveDomainByRestQuery(nameHash);
-    } catch (err) {
-    // eslint-disable-next-line no-console
-      console.log(err);
-      throw new Error('Failed to resolveSLD');
-    }
-  };
+    throw new Error(`SLD message for:[${nameHash.domain}] not found on topic:[${tld.topicId.toString()}]`);
```

Comment:
same comment about the errors here This would help people implementing the sdk customize how they want to handle errors in specific situations.